packages/runtime/runtime-utils/src/test/serializer.spec.ts

```diff
@@ -49,19 +71,63 @@ describe("FluidSerializer", () => {
                 assert.strictEqual(actual, input,
                     "replaceHandles() on input with no handles must return original input.");
 
+                const decoded = serializer.decode(actual);
+                assert.strictEqual(decoded, input,
+                    "decode() on input with no handles must return original input.");
+                assert.deepStrictEqual(decoded, input,
+                    "input must round-trip through decode(replaceHandles()).");
+
                 const stringified = serializer.stringify(input, handle);
+                // Paranoid check that serializer.stringify() and JSON.stringify() agree.
+                assert.deepStrictEqual(stringified, JSON.stringify(input),
+                    "stringify() of input without handles must produce same result as JSON.stringify().");
+
```

Comment:
Serializer tests work by deterministically generating various object trees and verifying they round-trip as expected through encode/decode and stringify/parse.

There are several additional self-consistency checks, such as mixing encode () with the built-in JSON stringifier produces the same result as Serializer.stringify().  In the end, we have a strong case for correctness and consistency with JSON.